pkg/services/login/authinfo_test.go

```diff
@@ -0,0 +1,216 @@
+package login
+
+import (
+	"testing"
+
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsExternallySynced(t *testing.T) {
```

Comment:
I made an exhaustive list of tests. could be made smaller and only test one/two providers